src/openrct2-ui/windows/Network.cpp

```diff
@@ -0,0 +1,510 @@
+}
+
+#pragma region Information page
+
+static void window_network_information_mouseup(rct_window* w, rct_widgetindex widgetIndex)
+{
+    switch (widgetIndex)
+    {
+        case WIDX_CLOSE:
+            window_close(w);
+            break;
+        case WIDX_TAB1:
+            if (w->page != widgetIndex - WIDX_TAB1)
+            {
+                window_network_set_page(w, widgetIndex - WIDX_TAB1);
+            }
+            break;
+    }
+}
+
+static void window_network_information_resize(rct_window* w)
+{
+    window_set_resize(w, WW, WH, WW * 4, WH * 4);
+    window_network_anchor_border_widgets(w);
+}
+
+static void window_network_information_update(rct_window* w)
+{
+    w->frame_no++;
+    widget_invalidate(w, WIDX_TAB1 + w->page);
+    window_invalidate(w);
+
+    NetworkStats_t curStats = network_get_stats();
+
+    uint32_t currentTicks = platform_get_ticks();
+
+    float graphTimeElapsed = (currentTicks - _lastGraphUpdateTime) / 1000.0f;
+    _lastGraphUpdateTime = currentTicks;
+
+    for (int i = 0; i < NETWORK_STATISTICS_GROUP_MAX; i++)
+    {
+        uint32_t deltaBytesReceived = curStats.bytesReceived[i] - _networkStats.bytesReceived[i];
+        uint32_t deltaBytesSent = curStats.bytesSent[i] - _networkStats.bytesSent[i];
+
+        _networkLastDeltaStats.deltaBytesReceived[i] = deltaBytesReceived;
+        _networkLastDeltaStats.deltaBytesSent[i] = deltaBytesSent;
+
+        _networkAccumulatedStats.deltaBytesReceived[i] += deltaBytesReceived;
+        _networkAccumulatedStats.deltaBytesSent[i] += deltaBytesSent;
+    }
+
+    float graphMaxIn = 0.0f;
+    float graphMaxOut = 0.0f;
+
+    for (size_t i = 0; i < _networkHistory.size(); i++)
+    {
+        const NetworkHistory_t& history = _networkHistory[i];
+        for (int n = 1; n < NETWORK_STATISTICS_GROUP_MAX; n++)
+        {
+            graphMaxIn = (float)std::max<uint32_t>(history.deltaBytesReceived[n], graphMaxIn);
+            graphMaxOut = (float)std::max<uint32_t>(history.deltaBytesSent[n], graphMaxOut);
+        }
+    }
+
+    _graphMaxIn = flerp(_graphMaxIn, graphMaxIn, graphTimeElapsed * 4.0f);
+    _graphMaxOut = flerp(_graphMaxOut, graphMaxOut, graphTimeElapsed * 4.0f);
+
+    // Compute readable statistics.
+    if (currentTicks - _lastStatsUpdateTime >= 1000)
+    {
+        float statsTimeElapsed = (currentTicks - _lastStatsUpdateTime) / 1000.0f;
+        _lastStatsUpdateTime = currentTicks;
+
+        _bytesIn = _networkAccumulatedStats.deltaBytesReceived[NETWORK_STATISTICS_GROUP_TOTAL];
+        _bytesOut = _networkAccumulatedStats.deltaBytesSent[NETWORK_STATISTICS_GROUP_TOTAL];
+        _bytesInSec = (double)_bytesIn / statsTimeElapsed;
+        _bytesOutSec = (double)_bytesOut / statsTimeElapsed;
+
+        _networkAccumulatedStats = {};
+    }
+
+    _networkStats = curStats;
+    _networkHistory.push_back(_networkLastDeltaStats);
+}
+
+static void format_readable_size(char* buf, size_t bufSize, uint64_t bytes)
```

Comment:
I think this function should go into Localisation.cpp, so other windows can make use of it easily as well.

Reply:
Alrite I'll see how it fits in.

Reply:
Fits in very well :)